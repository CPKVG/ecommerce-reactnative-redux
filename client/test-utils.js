// import { configureStore } from '@reduxjs/toolkit';
// import React from 'react';
// import { render as rtlRender } from '@testing-library/react-native'
// import { Provider } from 'react-redux';
// import { PersistGate } from 'redux-persist/integration/react';
// import { persistor } from './redux/createStore';
// import { rootReducer } from './redux/rootReducer';

// //custom render

// function render(
//     ui ,{
//         preloadedState,
//         store = configureStore({ reducer: rootReducer, preloadedState}),
//         ...renderOptions
//     } = {}
// ){
//   function Wrapper( {childern} ){
//       return <Provider store = {store}>
//           <PersistGate persistor={persistor}>
//             {childern}
//           </PersistGate>
//       </Provider>
//   }  
//   return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
// }

// //re-export everything
// export * from '@testing-library/react-native'
// //override render method
// export { render }


import React from "react"
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { rootReducer } from "./redux/rootReducer";

const INITIAL_STATE = { ...rootReducer }; // Or use value from root reducer
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
// const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    
    wrapper: ({ children }) => <
      Provider store={store}>
          {children}
      </Provider>,
    ...options
  });

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender as render, store };