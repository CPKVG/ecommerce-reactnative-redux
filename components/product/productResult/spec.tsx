import React from "react";
import { cleanup, fireEvent } from "@testing-library/react-native";
import ProductResult from ".";
import { NavigationContainer } from '@react-navigation/native';
import renderer from "react-test-renderer";
import { FlatList, Image, Text, TouchableOpacity, View } from "react-native";
import ProductImage from "../productImage";
import { render, store } from "../../../test-utils";
import { shallow } from "enzyme";

jest.useFakeTimers()

afterEach(cleanup)

describe('productResult component', () => {
    const component = (
        <NavigationContainer>
            <ProductResult/>
        </NavigationContainer>
    )

    const component1 = (
      <View>
        <Text testID="titleID">
        titleID
        </Text>
      </View>
    )
      
    const productDataSearch = [{
        category:"Apparel",
        description:"Keeps you warm",
        id:"0",
        image:"White_t-shirt",
        price:"5.99",        
        stock:"2",
        title:"White T-shirt",
    }]

    test('should render', () => { 
        expect(component).not.toBe(null)
      })


      //TODO, testing flatlist on react-native seems too difficult

      // it('should flatlist return keyExtractor correctly', () => {
      //   const renderItem = jest.fn();
      //   const { getByTestId  } = render(component, {    
      //     productDataSearch
      //     })
      //     fireEvent.press(getByTestId('flat-list'));
      //     expect(renderItem).toHaveBeenCalled();
       
      //  });


      // it("should render product search results",()=>{
      //   const { getByText } = render(component, {    
      //     productDataSearch
      //     })
      //     expect(getByText("White T-shirt")).toBeTruthy();
      // })

    // it("should render product search results",()=>{
    //     const { getByTestId } = render(component, {    
    //       productDataSearch
    //       })
    //       // expect(getByTestId("imageID")).toBeTruthy();
    //       expect(getByTestId("titleID")).toBeTruthy();
    //       // expect(getByTestId("priceID")).toBeTruthy();
    // })
    

      
  it("renders the productDetail component via snapshot",()=> {
    const renderItem = ({item}:any) => {
    
        return(
          //<ProductImage/>
          <TouchableOpacity>
            <Image 
              source = {ProductImage(item.image)}
            />
            <Text>
              {item.title}
            </Text>
            <Text>
              ${item.price} nzd
            </Text>
          </TouchableOpacity >
        )
      }
    const tree = renderer.create(
        <View>
        <FlatList  
        data={productDataSearch}
        numColumns ={2}
        ItemSeparatorComponent={
            () => <View style={{ width: 16 }}/>
        }
        renderItem={renderItem}
  
    />
  
      </View>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

})