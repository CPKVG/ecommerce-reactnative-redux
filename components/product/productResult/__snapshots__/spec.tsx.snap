// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productResult component renders the productDetail component via snapshot 1`] = `
<View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "category": "Apparel",
          "description": "Keeps you warm",
          "id": "0",
          "image": "White_t-shirt",
          "price": "5.99",
          "stock": "2",
          "title": "White T-shirt",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assests/img/White_t-shirt.png",
                }
              }
            />
            <Text>
              White T-shirt
            </Text>
            <Text>
              $
              5.99
               nzd
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
